api/clients/v2/disperser_client.go

```diff
@@ -128,42 +145,67 @@ func (c *disperserClient) DisperseBlob(
 	blobVersion corev2.BlobVersion,
 	quorums []core.QuorumID,
 ) (*dispv2.BlobStatus, corev2.BlobKey, error) {
-	c.requestMutex.Lock()
-	defer c.requestMutex.Unlock()
+	return c.DisperseBlobWithProbe(ctx, data, blobVersion, quorums, nil)
+}
 
-	err := c.initOnceGrpcConnection()
+// DisperseBlobWithProbe disperses a blob with the given data, blob version, and quorums. If sequenceProbe is not nil,
+// the probe is used to capture metrics during the dispersal process.
+func (c *disperserClient) DisperseBlobWithProbe(
+	ctx context.Context,
+	data []byte,
+	blobVersion corev2.BlobVersion,
+	quorums []core.QuorumID,
+	probe *common.SequenceProbe,
+) (*dispv2.BlobStatus, corev2.BlobKey, error) {
+
+	if len(quorums) == 0 {
+		return nil, [32]byte{}, api.NewErrorInvalidArg("quorum numbers must be provided")
+	}
+	if c.signer == nil {
+		return nil, [32]byte{}, api.NewErrorInternal("uninitialized signer for authenticated dispersal")
+	}
+	for _, q := range quorums {
+		if q > corev2.MaxQuorumID {
+			return nil, [32]byte{}, api.NewErrorInvalidArg("quorum number must be less than 256")
+		}
+	}
+
+	probe.SetStage("acquire_accountant_lock")
+	c.accountantLock.Lock()
+
+	probe.SetStage("prepare_for_dispersal")
+
+	err := c.initOncePopulateAccountant(ctx)
 	if err != nil {
 		return nil, [32]byte{}, api.NewErrorFailover(err)
 	}
-	err = c.initOncePopulateAccountant(ctx)
+	err = c.initOnceGrpcConnection()
```

Comment:
Why are you making this `initOnceGrpcConnection` call while you have the accountant lock? It seems like you'd want to do this outside the lock.

Side note, `initOnceGrpcConnection` is also called within `initOncePopulateAccountant`, so the call in its current place isn't *actually* doing anything, and would need to be placed prior to acquiring the lock if we want the grpc connection initialized outside the lock.

Reply:
Moved above accountant lock.